File: Labs/lab01/jxk3039_lab01.py
# Jacob Ketterer
# 1001793039
# Python Version 3.11.5
# OS: Windows

import os
def main():
    total = 0
    total = dirspace(".")
    print(total)
def dirspace(path):
    sum = 0
    #for every item in the listed directory, create a new path for the child, and check file type
    #if the new path for the child is a directory, call recursively, else add to sum
    for item in os.listdir(path):
        newpath = os.path.join(path,item)
        if os.path.isfile(newpath):
            sum+=os.path.getsize(newpath)
        if os.path.isdir(newpath):
            sum+=dirspace(newpath)
    return sum
main()

File: Labs/lab01/jxk3039_lab01.java
// Jacob Ketterer
// 1001793039
// java version "17.0.4.1" 2022-08-18 LTS
// OS: Windows

import java.io.*;

public class jxk3039_lab01
{
    public static int dirspace(File path)
    {
        int sum = 0;
        //creates a filearray of every new path to children of the "File path" input
        File[] filearray = path.listFiles();
        for(int i=0;i<filearray.length;i++)
        {
            if(filearray[i].isDirectory())
                sum+=dirspace(filearray[i]);
            if(filearray[i].isFile())
                sum+=(filearray[i].length());
        }
        return sum;
    }
    public static void main(String[] args)
    {
        File path = new File(".");
        int total = dirspace(path);
        System.out.print(total);
    }
}